openapi-cli/src/main/java/io/ballerina/openapi/cmd/BallerinaCodeGenerator.java

```diff
@@ -263,12 +261,13 @@ public void generateClient(String definitionPath, String outPath, Filter filter,
      * @throws BallerinaOpenApiException when code generator fails
      */
     public void generateService(String definitionPath, String serviceName, String outPath, Filter filter,
```

Comment:
This method has a lot of boolean options, we can introduce a record for this.
I did something similar in here: https://github.com/ballerina-platform/openapi-tools/blob/f7a5faedcb2fc28910dd24a89028c809336ae0bd/openapi-cli/src/main/java/io/ballerina/openapi/cmd/BallerinaCodeGenerator.java#L290

Reply:
I think its the same function. Maybe we can introduce this in your PR. Otherwise it will be conflicting right

Reply:
I see, but we are using it in the client generation function as well right? Lets do this change in this PR for all the places this is applicable. I'll resolve the conflict in my PR when merging these changes

Reply:
fixed